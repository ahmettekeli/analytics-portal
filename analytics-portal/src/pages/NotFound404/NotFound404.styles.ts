import styled from "styled-components";
import { StyledMotionDiv } from "pages/Overview/Overview.styles";

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100%;
  width: 100%;
  margin: auto;
`;

export { StyledMotionDiv };
