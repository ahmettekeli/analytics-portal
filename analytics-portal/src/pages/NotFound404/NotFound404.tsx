import { Wrapper } from "./NotFound404.styles";

function NotFound404() {
  return (
    <Wrapper data-testid="NotFound404">
      <h1>404</h1>
      <h2>Page not found</h2>
    </Wrapper>
  );
}

export default NotFound404;
