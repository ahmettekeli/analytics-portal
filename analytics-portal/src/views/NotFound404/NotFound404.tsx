import { motion } from "framer-motion";
import { Wrapper } from "./NotFound404.styles";

function NotFound404() {
  return (
    <motion.div
      initial={{ opacity: 0 }}
      animate={{ opacity: 1 }}
      exit={{ opacity: 0 }}
    >
      <Wrapper>
        <h1>404</h1>
        <h2>Page not found</h2>
      </Wrapper>
    </motion.div>
  );
}

export default NotFound404;
