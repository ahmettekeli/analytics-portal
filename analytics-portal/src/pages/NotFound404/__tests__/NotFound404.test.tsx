import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import NotFound404 from "../NotFound404";

describe("NotFound404", () => {
  test("Renders NotFound404 component", () => {
    const renderResult = render(<NotFound404 />);
    expect(renderResult.getByTestId("NotFound404")).toBeInTheDocument();
  });

  test("NotFound404 matches snapshot", () => {
    const tree = renderer.create(<NotFound404 />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
