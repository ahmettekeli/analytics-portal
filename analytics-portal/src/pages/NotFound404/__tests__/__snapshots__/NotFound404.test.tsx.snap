// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound404 NotFound404 matches snapshot 1`] = `
<div
  className="sc-bdvvtL fsAlg"
  data-testid="NotFound404"
>
  <h1>
    404
  </h1>
  <h2>
    Page not found
  </h2>
</div>
`;
